configure.ps1

```diff
@@ -34,50 +31,139 @@ Param (
 
 . "$PSScriptRoot\build\common.ps1"
 
-Trace-Log "Configuring NuGet.Client build environment for VS${ToolsetVersion} Toolset"
+Trace-Log "Configuring NuGet.Client build environment"
+
+$BuildErrors = @()
 
-Update-SubModules -Force:$Force
+Invoke-BuildStep 'Configuring git repo' {
+    Update-SubModules -Force:$Force
+} -ev +BuildErrors
 
-Install-NuGet -Force:$Force
+Invoke-BuildStep 'Installing NuGet.exe' {
+    Install-NuGet -Force:$Force
+} -ev +BuildErrors
 
-Install-DotnetCLI -Force:$Force
+Invoke-BuildStep 'Installing .NET CLI' {
+    Install-DotnetCLI -Force:$Force
+} -ev +BuildErrors
 
-if ($CleanCache) {
+# Restoring tools required for build
+Invoke-BuildStep 'Restoring solution packages' {
```

Comment:
This should be done in `build.ps1` still. There are packages used by the build, i.e. `Microsoft.VisualStudio.ProjectSystem`. Additionally, all the other restores are happening there 😄